cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -66,10 +71,12 @@ func NewZipDeployClient(
 	// We do not have a Resource provider to register
 	options.DisableRPRegistration = true
 
-	// Increase default retry attempts from 3 to 4 as zipdeploy often fails with 3 retries.
+	// When using default retry (3), increase its attempts from 3 to 4 as zipdeploy often fails with 3 retries.
 	// With the default azcore.policy options of 800ms RetryDelay, this introduces up to 20 seconds of exponential back-off.
-	options.Retry = policy.RetryOptions{
-		MaxRetries: 4,
+	if options.Retry.MaxRetries == 3 {
```

Comment:
Curious about the logic here to *only* increase to 4 if the retry attempt is 3. I wonder if it's better to always set it to some retry attempt we care about. It's okay even if the options passed in had a higher number of attempts, we just want to ensure we know how long our request executes.

Reply:
I think the code here should inherit default settings from the application, and apply specific settings it cares about. Retries is one of those settings it cares about, and it should have complete control over the desired value. 

I also now notice that we reverted the [previous change](https://github.com/Azure/azure-dev/pull/3965/commits/bb7c9327d9e3703696046e32e78a581ea1f3cd13), and this is now modifying the passed in `*arm.ClientOptions` which is unexpected.